[Feat] Calendar 로직 및 테스트 코드 구현

seokzin opened the pull request:
## 📌 이슈

- close #142

<br/>

## 📝 설명

### getTimeFormat(date)
- 시간 포맷을 객체 형태로 리턴

### generateCalendar(date)
- 날짜를 입력 받으면 배열에 달력 객체를 담아 리턴

<br/>

## 📷 스크린샷
이런 식으로 담아줘서 뷰 로직에서 쉽게 사용 가능해요.
```typescript
[
      {
        year: 2022,
        month: 7,
        date: 28,
        today: false,
        prev: true,
        next: false
      },
      {
        year: 2022,
        month: 7,
        date: 29,
        today: false,
        prev: true,
        next: false
      },
      {
        year: 2022,
        month: 7,
        date: 30,
        today: false,
        prev: true,
        next: false
      },
      {
        year: 2022,
        month: 7,
        date: 31,
        today: false,
        prev: true,
        next: false
      },
      {
        year: 2022,
        month: 8,
        date: 1,
        today: false,
        prev: false,
        next: false
      },
      {
        year: 2022,
        month: 8,
        date: 2,
        today: false,
        prev: false,
        next: false
      },

   ...

     {
        year: 2022,
        month: 8,
        date: 29,
        today: false,
        prev: false,
        next: false
      },
      {
        year: 2022,
        month: 8,
        date: 30,
        today: false,
        prev: false,
        next: false
      },
      {
        year: 2022,
        month: 9,
        date: 1,
        today: false,
        prev: false,
        next: true
      }
]
```

<br/>

seokzin committed:
feat: getTimeFormat 함수 구현

seokzin committed:
test: 인자 생략시 기본 값인 현재 시간 리턴

seokzin committed:
docs: JSdoc 함수 설명 추가

devpla reviewed (approved):
직관적이고 깔끔하네요..! LGTM 👏

seokzin committed:
feat: jest types 라이브러리 설치

seokzin committed:
feat: 반환 값에 '요일' 추가

seokzin committed:
test: 테스트 요소 간소화

- 테스트에서 new Date()를 2번 호출하는 동안 1ms 오차 발생
- 연/월/일만 테스트 하도록 간소화
- 아주 낮은 확률로 사이드이펙트 존재 ('일'이 바뀔 때)

seokzin committed:
chore: jest watch 스크립트 추가

seokzin committed:
feat: 파라미터 기본 값 제거

- 활용성이 없음
- 비순수 함수의 순수 함수화

seokzin committed:
test: 테스트케이스 보강

seokzin committed:
test: getMonth의 시작 값이 0

seokzin committed:
test: time format 응용 사례

seokzin committed:
refac: 중복된 변수명 분리

- scope 충돌은 없지만 가독성 개선 목적

seokzin committed:
feat: 달력 생성 함수 구현

seokzin committed:
refac: 변수명 구체화 및 불필요한 변수 제거

seokzin committed:
refac: JSDoc (용어 설명) 추가

seokzin committed:
test: 윤년 정상 출력 테스트

seokzin committed:
refac: 린터 수동 적용

seokzin committed:
fix: 린터 오류 해결

- 'React' refers to a UMD global, but the current file is a module
- 많은 방법 시도했으나 import 외엔 해결 불가

seokzin commented:
Storybook v7 migration

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-65x-to-700

seokzin committed:
feat: 스토리북 빌드 파일 무시

seokzin committed:
refac: any 타입 제거
